src/main/java/org/jitsi/videobridge/rest/LongPollingServlet.java

```diff
@@ -38,7 +59,55 @@ protected void doGet(
         throws IOException,
                ServletException
     {
-        // TODO Auto-generated method stub
-        response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+        System.out.println("BB: IN LONG POLLING");
+        Continuation continuation = ContinuationSupport.getContinuation(request);
+        
+        // We don't get the target here (TODO: is that something that could
+        //  be fixed?), so we need to do a bit more work to extract the bits of the path
+        //  we're interested in.
+        // The stuff we're interested in starts after the /colibri/
+        String colibriPath = request.getRequestURL().toString().split("colibri/")[1];
+        String target = colibriPath.split("/")[0];
+
+        continuation.suspend();
+        Videobridge videobridge = getVideobridge();
+        System.out.println("got target: " + target);
+        if (target.equals("subscribeToEvents")) 
+        {
+          String confId = colibriPath.split("/")[1];
+          System.out.println("BB: subscribing to conference events for conference " + confId);
+          Conference conference = videobridge.getConference(confId, null);
+          if (conference == null)
+          {
+            System.out.println("Couldn't find conference " + confId);
+          }
+          else
+          {
+            System.out.println("found conf object");
+            System.out.println("Waiting for event");
+            try
+            {
+              JSONObject pushEvent = conference.getPushEvent();
```

Comment:
That's a blocking method call and is not the way to go with asynchronous processing (as commented above). What you want to do is suspend the request, register the suspension/continuation with another entity which monitors the blocking queue and resumes the continuation when at least one element is queued. When the request is resumed, read from the queue without blocking.